src/core/core.controller.js

```diff
@@ -627,8 +627,13 @@ module.exports = function(Chart) {
 			}
 
 			// On Hover hook
-			if (hoverOptions.onHover) {
-				hoverOptions.onHover.call(me, me.active);
+			if (e.type === 'mousemove') {
+			    if (hoverOptions.onHover) {
+			        hoverOptions.onHover.call(me, me.active);
+			    }
+			    if (me.legend && me.legend.handleEvent && me.legend.options.onHover) {
```

Comment:
Because if the user hasn't initialized the function onHover in their chart cofig, we won't need to go through the whole process of calculating the mouse position and if it is within the boundaries of the legend hitboxes. For the onClick event, it doesnt matter so much because it's activated once, but on the onHover it will be executed continuously so with that check I am not allowing the code to propagate to calling the legend handleEvent function. If I allow the code to propagate to the handleEvent function it will do all those mouse coordinate calculations, go through the forloop checking each and every legend hitbox and if the mouse is in it to eventually find that onHover is null and not do anything.